src/background/utils/options.js

```diff
@@ -98,6 +120,9 @@ export function setOption(key, value) {
       optionValue = objectSet(getOption(mainKey), keys.slice(1), value);
     }
     options[mainKey] = optionValue;
+    if (key === 'scriptTemplate') {
+      options[`${key}Edited`] = !!value && value !== defaults[key];
+    }
```

Comment:
What about calculate this value in options page and pass it back?
I don't think such `if`-condition in a common function is a good idea. 😅

Reply:
There are two places where `setOption` is called for this value (vm-template.vue and background/index.js). We would have to check in both places and remember to add a check to each such new place in the future so I decided it's too fragile.

Reply:
Also, if we do it elsewhere, we would have to run an additional `setOption`, which is kinda wasteful in this case as the option is just an internal indicator that doesn't need to be broadcast.

Reply:
We can use a declarative approach:
```js
const onSet = {
  scriptTemplate: (value, options) => { .... },
}
```
Then in setOption just before browser.storage.local.set:
```js
const hook = onSet[key];
if (hook) hook(optionValue, options);
```

Reply:
There is something similar:

https://github.com/violentmonkey/violentmonkey/blob/c0a7ec8256f452fb83f2bbf947b534eb2e86e5e0/src/background/index.js#L28-L33

Reply:
Ah, I didn't realize I can hijack this. So you're not concerned with the unnecessary processing and broadcasting of two options due to two setOption calls? Or maybe we can add a noBroadcast parameter?

Reply:
The hook is debounced so it won't be called frequently. The `UpdateOptions` message is only posted to options page and popup page, so I won't worry about broadcasting.
I don't think `noBroadcast` is a good idea because it may lead to difference between data in different contexts.

Reply:
Done.